src/librustc_trans/abi.rs

```diff
@@ -133,32 +134,279 @@ impl ArgAttributes {
     }
 }
 
+pub trait CastTarget {
```

Comment:
I wanted to make sure it works first, and the trait allows replacing with an enum later, but maybe I should do it now.